internal/controller/buildpipeline/buildpipeline_adapter.go

```diff
@@ -558,6 +558,14 @@ func (a *Adapter) prepareSnapshotForPipelineRun(pipelineRun *tektonv1.PipelineRu
 		snapshot.Labels[gitops.BuildPipelineRunFinishTimeLabel] = strconv.FormatInt(time.Now().Unix(), 10)
 	}
 
+	if gitops.IsSnapshotCreatedByPACMergeQueueEvent(snapshot) {
+		pullRequestNumber := gitops.ExtractPullRequestNumberFromMergeQueueSnapshot(snapshot)
+		if pullRequestNumber != "" {
+			snapshot.Labels[gitops.PipelineAsCodePullRequestAnnotation] = pullRequestNumber
```

Comment:
Just out of curiosity, why do we populate both annotation and labels with the same information?

Reply:
Since this is a key piece of information, and PaC has moved labels to annotations in the past, I wanted to ensure we put the info in both places to reduce risk of breaking changes in the future.